File: frontend_flutter_backup/lib/repositories/social_repository.dart
import '../models/social_post.dart';
import '../services/api_service.dart';

class SocialRepository {
  SocialRepository(this.api);

  final ApiService api;

  Future<List<SocialPost>> feed() async {
    final response = await api.dio.get('/social/feed');
    return (response.data as List? ?? const [])
        .whereType<Map>()
        .map((item) => SocialPost.fromJson(item.cast<String, dynamic>()))
        .toList();
  }

  Future<void> createPost({
    required String text,
    required String location,
    String? imageUrl,
  }) {
    return api.dio.post<void>(
      '/social/posts',
      data: {
        'text': text,
        'location': location,
        if (imageUrl != null) 'imageUrls': [imageUrl],
      },
    );
  }

  Future<void> toggleLike(String postId) =>
      api.dio.post<void>('/social/posts/$postId/like');

  Future<void> addComment(String postId, String text) => api.dio.post<void>(
    '/social/posts/$postId/comments',
    data: {'text': text},
  );

  Future<void> toggleFollow(String userId) =>
      api.dio.post<void>('/social/users/$userId/follow');

  Future<String> openDirectConversation(String userId) async {
    final response = await api.dio.post<dynamic>(
      '/conversations/direct/$userId',
    );
    final data = (response.data as Map).cast<String, dynamic>();
    return data['id']?.toString() ?? userId;
  }
}

File: frontend/src/repositories/socialRepository.js
import { api } from '../services/api';
import { mapSocialPost } from '../models';

export const socialRepository = {
  async feed() {
    const { data } = await api.get('/social/feed');
    return (data || []).map(mapSocialPost);
  },

  async createPost({ text, location, imageUrl }) {
    await api.post('/social/posts', {
      text,
      location,
      ...(imageUrl ? { imageUrls: [imageUrl] } : {}),
    });
  },

  async toggleLike(postId) {
    await api.post(`/social/posts/${postId}/like`);
  },

  async addComment(postId, text) {
    await api.post(`/social/posts/${postId}/comments`, { text });
  },

  async toggleFollow(userId) {
    const { data } = await api.post(`/social/users/${userId}/follow`);
    return data;
  },
};
